integrations/consul/proxysql-consul

```diff
@@ -0,0 +1,211 @@
+TABLE_TO_KEY = {
+        'mysql_servers': 'proxysql/mysql_servers',
+        'mysql_query_rules': 'proxysql/mysql_query_rules'
+        }
+
+SAVE_QUERY_BY_TABLE = {
+        'mysql_servers': 'SAVE MYSQL SERVERS TO MEMORY',
+        'mysql_query_rules': 'SAVE MYSQL QUERY RULES TO MEMORY'
+        }
+
+LOAD_QUERY_BY_TABLE = {
+        'mysql_servers': 'LOAD MYSQL SERVERS FROM MEMORY',
+        'mysql_query_rules': 'LOAD MYSQL QUERY RULES FROM MEMORY'
+        }
+
+config = {}
+
+
+def read_config():
+    global config
+    with open(CFG_PATH) as config_file:
+        config_data = config_file.read()
+    config = json.loads(config_data)
+
+    # TODO validate config
+
+
+def read_uuid():
+    """
+    Reads the UUID identifying this script instance from a file store. If the
+    file does not exist creates it and writes a new UUID.
+
+    Puts the UUID in the config dictionary.
+    """
+    uuid_file_path = config[CFG_UUID_FILE]
+
+    if not os.path.isfile(uuid_file_path):
+        _uuid = str(uuid.uuid4())
+        with open(uuid_file_path, 'w') as uuid_file:
+            uuid_file.write(_uuid)
+    else:
+        with open(uuid_file_path, 'r') as uuid_file:
+            _uuid = uuid_file.read()
+    config[CFG_UUID] = _uuid
+
+def fetch_proxysql_config(table):
+    admin_connection = MySQLdb.connect(config[CFG_PROXY_IFACE],
+            config[CFG_PROXY_USERNAME],
+            config[CFG_PROXY_PASSWORD],
+            port=config[CFG_PROXY_PORT],
+            db='main')
+
+    # save runtime config to memory so we can read them
+    cursor = admin_connection.cursor()
+    save_query = SAVE_QUERY_BY_TABLE[table]
+    cursor.execute(save_query)
+    cursor.close()
+
+    # read runtime config from memory
+    cursor = admin_connection.cursor()
+    cursor.execute('SELECT * FROM %s' % table)
+    rows = cursor.fetchall()
+    cursor.close()
+    admin_connection.close()
+    return rows 
+
+
+def put_config(table):
+    if table not in TABLE_TO_KEY:
+        print 'Unknown config table. Exiting.'
+        exit(1)
+
+    rows = fetch_proxysql_config(table)
+    consul_data = {}
+    consul_data['table'] = table
+    consul_data['rows'] = rows
+    consul_data['uuid'] = config[CFG_UUID]
+    consul_data_json = json.dumps(consul_data)
+
+    key = TABLE_TO_KEY[table]
+    consul_iface = config[CFG_CONSUL_IFACE]
+    consul_port = config[CFG_CONSUL_PORT]
+
+    url = 'http://%s:%s/v1/kv/%s' % (consul_iface, consul_port, key)
+    r = requests.put(url, data=consul_data_json)
+    print 'Configs pushed successfully.'
+
+
+def build_multivalue_insert(table, rows):
+    # add quotes arround all values to make them strings in the sql query
+    quoted_values = [['"' + x + '"' for x in row] for row in rows]
+    # join each row in a values() group
+    row_join = [','.join(x) for x in quoted_values]
+    query = 'INSERT INTO %s VALUES(%s)' % (table, '),('.join(row_join))
+    return query
+
+def update_proxysql_config(table, rows):
```

Comment:
`update_proxysql_runtime_config` is a bit more specific. Because it has 3 layers of config.